src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -794,6 +794,53 @@ class UseDatabaseStatementSegment(BaseSegment):
     )
 
 
+# Data Manipulation Statements
+@spark3_dialect.segment()
+class InsertTableStatement(BaseSegment):
+    """A `INSERT [TABLE]` statement to insert or overwrite new rows into a table.
+
+    https://spark.apache.org/docs/latest/sql-ref-syntax-dml-insert-into.html
+    https://spark.apache.org/docs/latest/sql-ref-syntax-dml-insert-overwrite-table.html
+    """
+
+    type = "insert_table_statement"
+
+    match_grammar = Sequence(
+        "INSERT",
+        OneOf("INTO", "OVERWRITE"),
+        Ref.keyword("TABLE", optional=True),
+        Ref("TableReferenceSegment"),
+        Ref("PartitionSpecGrammar", optional=True),
+        Ref("BracketedColumnReferenceListGrammar", optional=True),
+        OneOf(
+            AnyNumberOf(
+                Ref("ValuesClauseSegment"),
+                min_times=1,
+            ),
+            Ref("SelectableGrammar"),
+            Sequence(
+                Ref.keyword("TABLE", optional=True),
+                Ref("TableReferenceSegment"),
+            ),
+            Sequence(
+                # TODO is there a less brittle way to do this using an existing class(es)?
+                # TODO if not, how to best construct a new class for `SELECT` match?
+                "FROM",
+                Ref("TableReferenceSegment"),
+                "SELECT",
+                Delimited(
+                    Ref("ColumnReferenceSegment"),
+                ),
+                Ref("TableReferenceSegment", optional=True),
+                Ref("WhereClauseSegment", optional=True),
+                Ref("GroupByClauseSegment", optional=True),
+                Ref("OrderByClauseSegment", optional=True),
+                Ref("LimitClauseSegment", optional=True),
```

Comment:
Are these all relevant for an INSERT TABLE FROM clause? Or is that a simplified select that can't do GROUP BYs? Or ORDER BYs? OR LIMIT? I don't know the answer btw as don't use spark3 so just asking the question.

Reply:
yes, these are all relevant for an `INSERT TABLE FROM` clause. It doesn't behave as a simplified select